common/djangoapps/util/course.py

```diff
@@ -0,0 +1,49 @@
+"""
+Utility methods related to course
+"""
+import logging
+from django.conf import settings
+
+from opaque_keys.edx.keys import CourseKey
+import re
+
+log = logging.getLogger(__name__)
+
+
+def get_lms_link_for_about_page(request, course_key):
+    """
+    Returns the url to the course about page from the location tuple.
+    """
+
+    assert isinstance(course_key, CourseKey)
+
+    if settings.FEATURES.get('ENABLE_MKTG_SITE', False):
+        if not hasattr(settings, 'MKTG_URLS'):
+            log.exception("ENABLE_MKTG_SITE is True, but MKTG_URLS is not defined.")
+            return None
+
+        marketing_urls = settings.MKTG_URLS
+
+        # Root will be "https://www.edx.org". The complete URL will still not be exactly correct,
+        # but redirects exist from www.edx.org to get to the Drupal course about page URL.
+        about_base = marketing_urls.get('ROOT', None)
+
+        if about_base is None:
+            log.exception('There is no ROOT defined in MKTG_URLS')
+            return None
```

Comment:
This code feels weird to me for several reasons

1. If about_base is None, there's no exception raised, so log.exception doesn't do anything more than log.error.  
2. Why are we waiting until someone requests this view to validate our MKTG_URLS settings?
3. Is it valid to have a `MKTG_URLS` with no `'ROOT'` key?  If not, just do `marketing_urls['ROOT']` and let the KeyError crash the app.
4. `x.get('ROOT', None)` is the same as `x.get('ROOT')`.